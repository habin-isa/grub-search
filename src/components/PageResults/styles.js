import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 50px;
  height: 100vh;
`;
export const VenuesTitle = styled.div`
  font-size: 20px;
  color: #333;
  padding-bottom: 30px;
`;

export const Venues = styled.div`
  display: flex;
  flex-direction: column;
  font-size: 12px;
  cursor: pointer;
  width: 100%;
`;

export const SimilarVenuesTitle = styled.div`
  font-size: 20px;
  color: #666464;
  padding-bottom: 30px;
`;

export const SimilarVenues = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  color: #666464;
`;

export const Graph = styled.div`
  display: flex;
  width: 100%;
`;
