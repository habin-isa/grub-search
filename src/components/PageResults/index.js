import React from 'react';
import * as S from './styles';
import GraphContainer from '../GraphContainer';
import { array, object } from 'prop-types';

const PageResults = ({ venues, renderedVenues, initialData }) => {
  return (
    <S.Wrapper>
      <S.Venues>
        <S.VenuesTitle>Please select one venue to see similar spots</S.VenuesTitle>
        {venues.length === 0 ? '0 Results' : renderedVenues}
      </S.Venues>
      <S.Graph>
        {' '}
        <S.SimilarVenues>
          <S.SimilarVenuesTitle>Similar venues:</S.SimilarVenuesTitle>
          <S.Venues>
            {initialData.firstResponse === undefined
              ? 'No results for similar venues'
              : initialData.firstResponse.map((venue, i) => <div key={i}>{venue.id}</div>)}
          </S.Venues>
          <S.Venues>Graph powered by d3</S.Venues>
        </S.SimilarVenues>{' '}
        <GraphContainer initialData={initialData} />
      </S.Graph>
    </S.Wrapper>
  );
};

PageResults.propTypes = {
  venues: array,
  renderedVenues: array,
  initialData: object
};

PageResults.defaultProps = {
  venues: [],
  renderedVenues: [],
  initialData: {}
};

export default PageResults;
