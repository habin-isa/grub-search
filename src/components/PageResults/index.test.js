import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import PageResults from '.';

describe('PageResults component', () => {
  const props = {
    venues: [
      { id: 'Costa', group: '1' },
      { id: 'Costa', group: '1' },
      { id: 'Costa', group: '1' }
    ],
    similarVenues: [
      { id: 'Costa', group: '1' },
      { id: 'Costa', group: '1' },
      { id: 'Costa', group: '1' }
    ],
    renderedVenues: [],
    renderedSimilarVenues: [],
    stopChart: 0
  };
  it('should render', () => {
    const { container } = render(<PageResults {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with correct venues data', () => {
    shallow(<PageResults venues={props.venues} />);
  });
  it('renders with correct similarVenues data', () => {
    shallow(<PageResults similarVenues={props.similarVenues} />);
  });
});
