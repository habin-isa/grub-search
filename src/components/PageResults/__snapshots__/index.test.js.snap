// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageResults component should render 1`] = `
<div
  class="sc-AxjAm eLTCfx"
>
  <div
    class="sc-AxiKw jOUsNA"
  >
    <div
      class="sc-AxirZ epZuHH"
    >
      Please select one venue to see similar spots
    </div>
  </div>
  <div
    class="sc-AxgMl ihzjTA"
  >
     
    <div
      class="sc-AxhUy gHEPPx"
    >
      <div
        class="sc-AxhCb jCFzct"
      >
        Similar venues:
      </div>
      <div
        class="sc-AxiKw jOUsNA"
      />
    </div>
     
    <div
      class="sc-AxheI iAyAiX"
    >
      <div
        class="chartData"
      >
        <svg
          viewBox="0,0,400,400"
        >
          <g
            stroke="#999"
            strokeOpacity="0.6"
          >
            <line
              strokeWidth="500"
            />
            <line
              strokeWidth="500"
            />
          </g>
          <g
            stroke="#fff"
            strokeWidth="1.5"
          >
            <circle
              fill="#1f77b4"
              r="10"
            />
            <circle
              fill="#1f77b4"
              r="10"
            />
            <circle
              fill="#1f77b4"
              r="10"
            />
          </g>
          <text
            dx="12"
            dy="0.35em"
          />
          <text
            dx="12"
            dy="0.35em"
          />
          <text
            dx="12"
            dy="0.35em"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
